/*
 * If not stated otherwise in this file or this component's LICENSE file the
 * following copyright and licenses apply:
 *
 * Copyright 2022 Sky UK
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 * http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

syntax = "proto2";

import "google/protobuf/descriptor.proto";
import "rialtoipc.proto";

package example;

// You need this to generate the rpc service stubs
option cc_generic_services = true;


message RequestEcho {
  required string text = 1;
}
message ResponseEcho {
  required string text = 1;
}


message RequestWithFd {
  optional string text = 1;
  required int32 fd = 2 [(rialto.ipc.field_is_fd) = true];
}
message ResponseWithFd {
  optional string text = 1;
  required int32 fd = 2 [(rialto.ipc.field_is_fd) = true];
}


message RequestWithNoReply {
  optional string text = 1;
}
message EmptyResponse {
}



message SomeEvent {
  required int32 id = 5;
  optional string text = 1;
}

service ExampleService {

  rpc exampleEcho(RequestEcho) returns (ResponseEcho) {
  }

  rpc exampleWithFd(RequestWithFd) returns (ResponseWithFd) {
  }

  rpc exampleWithNoReply(RequestWithNoReply) returns (EmptyResponse) {
    option (rialto.ipc.no_reply) = true;
  }

}
